### PR TITLE
Propose Traits::Access -> AccessTraits

### PR DESCRIPTION
**Pull request by dalg24**
Drop `Traits::` nested namespace.
Old syntax remains valid for now.

I intend on reverting 459023d before we merge.  It is mainly there to demonstrate that it works.

### CODE REVIEW

**Comment by aprokop** [comment]
@dalg24 Can you please explain the reasoning behind the change, what limitations/drawbacks the current version has, and directions for adding traits in the future?

**Comment by dalg24** [comment]
> @dalg24 Can you please explain the reasoning behind the change, what limitations/drawbacks the current version has, and directions for adding traits in the future?

* Nested namespace `Traits::` made thing awkward internally
  * closing `Details::` to open `Traits::`, then reopen `Details::` in `ArborX_AccessTraits.hpp`
  * having to qualify the tags in when using the traits
* Name is non descriptive.
* Also the only place in our API where we have a nested namespace.

No limitations.  No change in directions.

File: src/details/ArborX_AccessTraits.hpp [hunk 1]
```diff
@@ -90,93 +83,96 @@ template <typename Traits>
 using AccessTraitsGetArchetypeExpression = decltype(
     Traits::get(std::declval<first_template_parameter_t<Traits> const &>(), 0));
 
-} // namespace Details
-
-namespace Traits
-{
 template <typename Access>
 struct Helper
```
Comment (dalg24) [inline src/details/ArborX_AccessTraits.hpp:87]: We would need to rename this guy though.  Speaking of non descriptive names...

**Comment by dalg24** [comment]
Guess what compiler I was using...
I will figure out the errors with nvcc and pgi later if we agree we want that PR to go through.

**Comment by aprokop** [comment]
I agree that `Traits` should not be a namespace. And I agree that it should be part of `ArborX` namespace and not `ArborX::Details`, as it is part of user interactions.

**Pull request merged**